.github/workflows/create_workflows.py

```diff
@@ -0,0 +1,198 @@
+"""
+This scripts uses a jinja template to create CI workflows to test.
+
+    - different linux distributions (split by the package manager they use)
+    - different softwares that neurodocker supports
+    - different install method for a given software
+
+All of those are defined in a python dictionary.
+
+Versions to install are read from the neurodocker template for a given software.
+It is possible to skip a version by adding a "skip_versions" key to the software.
+
+Each workflow:
+
+    - installs the latest version of neurodocker
+    - builds a dockerfile for a combination of OS / software / version / install method
+    - cat the dockerfile
+    - attempts to build the corresponding image
+
+This script will also create a "dashboard" saved in docs/README.md
+to be picked up to be rendered by the github pages.
+This requires for you to build the pages from the docs folder
+and on the branch where the workflows are pushed to (currently "test_docker_build").
+
+"""
+from pathlib import Path
+
+import yaml  # type: ignore
+from jinja2 import Environment, FileSystemLoader, select_autoescape
+
+apt_based = [
+    "ubuntu:22.04",
+    "ubuntu:18.04",
+    "ubuntu:16.04",
+    "debian:bullseye-slim",
+    "debian:buster-slim",
+    "debian:stretch-slim",
+]
+yum_based = ["fedora:36", "centos:7"]
+
+"""
+Add a "skip_versions" key to the software dictionary if you want to skip
+testing a specific version. For example, if you want to skip testing
+version 1.0.0 of afni, add the following to the software dictionary:
+
+    "afni": {
+            "skip_versions": ["1.0.0"],
+            "methods": ["binaries", "source"],
+            "afni_python": ["true", "false"],
+        },
+
+"""
+softwares: dict[str, dict[str, list[str]]] = {
+    "afni": {
+        "methods": ["binaries", "source"],
+        "afni_python": ["true", "false"],
+    },
+    "ants": {
+        "methods": ["binaries", "source"],
+    },
+    "cat12": {"methods": ["binaries"]},
+    "convert3d": {"methods": ["binaries"]},
+    "dcm2niix": {
+        "methods": ["binaries", "source"],
+    },
+    "freesurfer": {"methods": []},
+    "fsl": {
+        "methods": ["binaries"],
+    },
+    "matlabmcr": {
+        "methods": ["binaries"],
+    },
+    "mricron": {"methods": ["binaries"]},
+    "mrtrix3": {
+        "methods": ["binaries", "source"],
+    },
+    "spm12": {"methods": ["binaries"]},
+}
+
+output_dir = Path(__file__).parent
+
+template_folder = Path(__file__).parents[2].joinpath("neurodocker", "templates")
+
+build_dashboard_file = Path(__file__).parents[2].joinpath("docs", "README.md")
```

Comment:
The dashboard is added to the docs folder and rendered with github pages.

That means that to make test this I had to nuke the `.nojekyll` file that was in there.